pkg/controllerutils/reconciler/operation_annotation_wrapper.go

```diff
@@ -34,22 +34,14 @@ type operationAnnotationWrapper struct {
 // removes the Gardener operation annotation before `Reconcile` is called.
 //
 // This is useful in conjunction with the HasOperationAnnotation predicate.
-func OperationAnnotationWrapper(newObjFunc func() client.Object, reconciler reconcile.Reconciler) reconcile.Reconciler {
+func OperationAnnotationWrapper(newObjFunc func() client.Object, mgr manager.Manager, reconciler reconcile.Reconciler) reconcile.Reconciler {
 	return &operationAnnotationWrapper{
+		client:     mgr.GetClient(),
```

Comment:
No strong opinion on this. Tried to streamline with other controllers. If in future the controller needs some option to be included from `manager`, it will be easier to adapt.